app/grandchallenge/components/views.py

```diff
@@ -244,6 +244,52 @@ def new_interface_url(self):
         raise NotImplementedError
 
 
+class CIVSetUpdateViewMixin(CIVSetFormMixin):
+    def process_data_for_object(self, data):
+        """Updates the object"""
+        instance = self.object
+        non_civ_data_keys = self.form_class.Meta.non_civ_fields
+
+        save = False
+        for key in non_civ_data_keys:
+            value = data.get(key, empty)
+            if value is not empty and value != getattr(instance, key):
+                setattr(instance, key, value)
+                save = True
+        if save:
+            instance.save()
+
+        for key in data:
+            if key in non_civ_data_keys:
+                continue
+            instance.create_civ(
+                ci_slug=key, new_value=data[key], user=self.request.user
+            )
+
+        return instance
+
+
+class CIVSetCreateViewMixin(CIVSetFormMixin):
+    def create_object(self, non_civ_data):
+        raise NotImplementedError
+
+    def process_data_for_object(self, data):
+        """Creates the object, including interfaces"""
+        non_civ_data_keys = self.form_class.Meta.non_civ_fields
+
+        instance = self.create_object(
+            non_civ_data={k: data.get(k) for k in non_civ_data_keys}
+        )
+
+        for key in data:
+            if key in non_civ_data_keys:
+                continue
+            instance.create_civ(
+                ci_slug=key, new_value=data[key], user=self.request.user
+            )

```

Comment:
```suggestion
            if key not in non_civ_data_keys:
                instance.create_civ(
                    ci_slug=key, new_value=data[key], user=self.request.user
                )

```

Reply:
This also seems like a duplicate fragment from `CIVSetFormMixin`, why duplicate rather than use inheritance?